pkg/resources/resources_manager_test.go

```diff
@@ -0,0 +1,290 @@
+			Expect(err.Error()).To(Equal("error: Invalid value for rdmaHcaMax < 0: -100"))
+		})
+		It("resources with no devices", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: no devices provided"))
+		})
+	})
+	Context("InitServers", func() {
+		It("Init valid server", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.InitServers()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(rm.configList)).To(Equal(1))
+		})
+		It("Init server with invalid number of resources", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   -100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.InitServers()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("StartAllServers", func() {
+		It("start valid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Start").Return(nil)
+			fakeDevicePlugin.On("Watch").Return(nil)
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.StartAllServers()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("start invalid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Start").Return(errors.New("failed"))
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.StartAllServers()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("StopAllServers", func() {
+		It("stop valid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Stop").Return(nil)
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.StopAllServers()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("stop invalid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Stop").Return(errors.New("failed"))
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.StopAllServers()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("RestartAllServers", func() {
+		It("restart valid server", func() {
+			fakeDevicePlugin := mocks.ResourceServer{}
+			fakeDevicePlugin.On("Restart").Return(nil)
+
+			rm := &resourceManager{resourceServers: []types.ResourceServer{&fakeDevicePlugin}}
+
+			err := rm.RestartAllServers()
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
you should evaluate that Restart() was called

Reply:
added new check `Expect(len(rm.resourceServers)).To(BeNumerically(">", 0))` to make sure it will go through the loop and run Restart()

Reply:
added check with AssertExpectations